src/engines/engine-decorator-provider.js

```diff
@@ -0,0 +1,24 @@
+// @flow
+
+/**
+ * Engine decorator provider.
+ * @class EngineDecoratorProvider
+ * @param {IEngineDecoratorProvider} plugin - The plugin which have implemented decorator.
+ * @implements {IEngineDecorator}
+ */
+class EngineDecoratorProvider implements IEngineDecoratorProvider {
+  constructor(plugin: IEngineDecoratorProvider) {
```

Comment:
this means we are still passing the entire plugin to playkit. lets pass only the decorator provider.

Reply:
we're passing it but not keeping it

Reply:
Provider will be getName and getEngineDecorator instead of the plugin itself